frontend/src/pages/Compare.test.tsx

```diff
@@ -14,647 +14,212 @@
+    expect(getRunSpy).toHaveBeenCalledWith(MOCK_RUN_2_ID);
+    expect(getRunSpy).toHaveBeenCalledWith(MOCK_RUN_3_ID);
   });
 
-  it("displays run's parameters if the run has any", async () => {
-    const workflow = {
-      spec: {
-        arguments: {
-          parameters: [
-            { name: 'param1', value: 'value1' },
-            { name: 'param2', value: 'value2' },
-          ],
-        },
-      },
-    } as Workflow;
-
-    const run = newMockRun('run-with-parameters');
-    run.pipeline_runtime!.workflow_manifest = JSON.stringify(workflow);
-    runs.push(run);
-
-    const props = generateProps();
-    props.location.search = `?${QUERY_PARAMS.runlist}=run-with-parameters`;
-
-    tree = shallow(<Compare {...props} />);
-    await TestUtils.flushPromises();
-    tree.update();
+  it('Show v1 page if all runs are v1 and the v2 feature flag is enabled', async () => {
+    const getRunSpy = jest.spyOn(Apis.runServiceApi, 'getRun');
+    runs = [
+      newMockRun(MOCK_RUN_1_ID, false),
+      newMockRun(MOCK_RUN_2_ID, false),
+      newMockRun(MOCK_RUN_3_ID, false),
+    ];
+    getRunSpy.mockImplementation((id: string) => runs.find(r => r.run!.id === id));
 
-    expect(tree.state('paramsCompareProps')).toEqual({
-      rows: [['value1'], ['value2']],
-      xLabels: ['test run run-with-parameters'],
-      yLabels: ['param1', 'param2'],
+    // v2 feature is turn on.
+    jest.spyOn(features, 'isFeatureEnabled').mockImplementation(featureKey => {
+      if (featureKey === features.FeatureKey.V2_ALPHA) {
+        return true;
+      }
+      return false;
     });
-    expect(tree).toMatchSnapshot();
-  });
-
-  it('displays parameters from multiple runs', async () => {
-    const run1Workflow = {
-      spec: {
-        arguments: {
-          parameters: [
-            { name: 'r1-unique-param', value: 'r1-unique-val1' },
-            { name: 'shared-param', value: 'r1-shared-val2' },
-          ],
-        },
-      },
-    } as Workflow;
-    const run2Workflow = {
-      spec: {
-        arguments: {
-          parameters: [
-            { name: 'r2-unique-param1', value: 'r2-unique-val1' },
-            { name: 'shared-param', value: 'r2-shared-val2' },
-          ],
-        },
-      },
-    } as Workflow;
-
-    const run1 = newMockRun('run1');
-    run1.pipeline_runtime!.workflow_manifest = JSON.stringify(run1Workflow);
-    const run2 = newMockRun('run2');
-    run2.pipeline_runtime!.workflow_manifest = JSON.stringify(run2Workflow);
-    runs.push(run1, run2);
 
-    const props = generateProps();
-    props.location.search = `?${QUERY_PARAMS.runlist}=run1,run2`;
-
-    tree = shallow(<Compare {...props} />);
-    await TestUtils.flushPromises();
-    tree.update();
+    render(
+      <CommonTestWrapper>
+        <Compare {...generateProps()} />
+      </CommonTestWrapper>,
+    );
 
-    expect(tree).toMatchSnapshot();
+    await waitFor(() => screen.getByText('Run overview'));
```

Comment:
Question: `getByText` is a synchronized function, so you don't need to `waitFor` it to return. That means you can directly call `screen.getByText('Run overview');` Does it make sense?

If you are trying to wait for something to show up, it is correct to use `waitFor`. But you need to combine with `queryByText`:  `await waitFor(async () => screen.queryByText('Run overview'));`

Reply:
Ok sounds good, no problem and thank you!